systemtest/src/main/java/io/strimzi/systemtest/performance/utils/TopicOperatorPerformanceUtils.java

```diff
@@ -0,0 +1,214 @@
+    private TopicOperatorPerformanceUtils() {}  // Prevent instantiation
+
+    /**
+     * Processes Kafka topic operations across multiple batches asynchronously.
+     * This method sets up a fixed thread pool based on the available CPU processors
+     * and manages batch operations for creation, modification, and deletion of Kafka topics.
+     *
+     * Illustration of Concurrent Batch Processing:
+     *
+     * ┌──────────────────────────────────────────────────────────────────────────────────────────────────┐
+     * │                                           Thread Pool                                            │
+     * │                                                                                                  │
+     * │   +------------------+    +------------------+    +------------------+    +------------------+   │
+     * │   | Thread 1         |    | Thread 2         |    | Thread 3         |    | Thread N         |   │
+     * │   | +--------------+ |    | +--------------+ |    | +--------------+ |    | +--------------+ |   │
+     * │   | | Batch 1      | |    | | Batch 2      | |    | | Batch 3      | |    | | Batch N      | |   │
+     * │   | |              | |    | |              | |    | |              | |    | |              | |   │
+     * │   | | 1. Creation  | |    | | 1. Creation  | |    | | 1. Creation  | |    | | 1. Creation  | |   │
+     * │   | | 2. Modific.  | | -->| | 2. Modific.  | | -->| | 2. Modific.  | | -->| | 2. Modific.  | |   │
+     * │   | | 3. Deletion  | |    | | 3. Deletion  | |    | | 3. Deletion  | |    | | 3. Deletion  | |   │
+     * │   | +--------------+ |    | +--------------+ |    | +--------------+ |    | +--------------+ |   │
+     * │   +------------------+    +------------------+    +------------------+    +------------------+   │
+     * │                                                                                                  │
+     * │ - Each thread processes a specific batch, performing creation, modification, and deletion        │
+     * │   sequentially, but each batch operation is independent and concurrent across threads.           │
+     * │ - Threads are managed by an executor service that efficiently schedules each batch               │
+     * │   task to maximize the available CPU resources.                                                  │
+     * │ - Errors in any batch operation are logged and do not impact the processing of other             │
+     * │   batches, ensuring robust execution.                                                            │
+     * └──────────────────────────────────────────────────────────────────────────────────────────────────┘
+     *
+     * @param testStorage    The storage instance containing the configuration and namespace details.
+     * @param numberOfTopics The total number of Kafka topics to process.
+     */
+    public static void processKafkaTopicBatchesAsync(TestStorage testStorage, int numberOfTopics) {
```

Comment:
Here you are batching in terms of topic resources. On the other end, the operator actually reconciles topic events, and this is what is measured by metrics.

A topic event has one of 2 types: `upsert` (create, update), `delete`. An update event can be further divided into 2 sub-types: `resync` (from periodic reconciliation), `update` (on-demand update).

Using `numberOfTopics` is fine, but you are actually generating a workload of `numberOfTopics * 3` events for the operator. This should be reflected on the X axis.

Reply:
Okay so what about adding a a `IN` metric with `NUMBER_OF_EVENTS`, which would be equal to `numberOfTopics` * 3?